### ISSUE TITLE
Request: Method to give existing channel funds to arbitrary account.

### ISSUE THREAD

**Issue by junderw** [BlueWallet/LndHub#192]
## Explanation:

1. On LND, I have X channels open with local balances of A, B, C.
2. I install LNDHub, create a wallet.
3. My wallet balance is 0 and the only way I can get balance on it is by "charging it" via BlueWallet on-chain, or receiving an LN payment.
4. LND won't let me pay an invoice created by itself, so the only method is on-chain "charging" which is a waste, since I already have `A + B + C` total LN balance available to me.

## Request:

1. On first boot of LNDHub from fresh slate, create a default user with a balance of 100% of all channels local balance at the time of first boot.
2. This user's backup can be shown for users to "load" the account into their BlueWallet.
3. The default user ID and the default user initial balance are persisted, and used to calculate the "true" balance of the default user moving forward. (so calculateBalance would add an if statement for if U.isDefaultUser() to add the instance's default user initial balance.)

## Some issues that might pop up:

How to keep the backup of that user secure? I would guess showing it only once is fine but tbh if you have access to the box with lndhub you can mess with redis and also probably swipe the admin macaroon anyways, so maybe just show the backup string of the default user whenever someone asks.

**Comment by Overtorment** [comment]
i talked to umbrel guys, and the simplest solution might be:
adding "admin" account credentials straight to config.js during install.

the balance would not be _awarded_ to it, instead, balance for this account would be simply current channelBalance, so admin user could spend all his channel coins anytime he wants.

what do you think?

**Comment by junderw** [comment]
As long as the admin credentials are randomly generated on install that sounds fine.
If it's the same credentials for everyone, then all I need is your onion URL to access your admin account.

But tbh if it is easy to add admin credentials on install, why not just add a one line command on install to add the current channel balance to the admin.

Why should you always keep the balance equal to channelBalance?

---

So on install:

1. Generate random admin account creds and save it to the lndhub volume folder (for docker)
2. Add entry to redis for:
  a. Admin account data (just like any other user)
  b. A new key called `admin_init_balance` which will be equal to current channelBalance at install time, a comma, then the UID of the admin account.

Then in lndhub logic:

On startup, some class should fetch the admin_init_balance and store it locally in memory as `this.adminInitBalance` and `this.adminUid` and these should be viewed in the getBalance function. (ie. `if (this.uid === globalThingy.adminUid) balance += globalThingy.adminInitBalance` and then the LND invoices etc will be subtracted/added to that just like any normal account)

**Comment by junderw** [comment]
... Well, yeah. I guess it would still not cover the following case:

1. User has no channels.
2. User installs lndhub.
3. User opens channels using umbrel / rtl etc. and gains X balance off-chain.
4. User installs admin account to bluewallet.

The channels opened in 3 won't be shown and we'll still have the same problem...

**Comment by junderw** [comment]
Yeah I guess it would be fine to just say that admin user's current balance === listchannels results localbalance sum.

**Comment by junderw** [comment]
As long as the admin credentials are randomly generated on install that sounds fine.
If it's the same credentials for everyone, then all I need is your onion URL to access your admin account.

**Comment by pinheadmz** [comment]
Joining this issue, I also am interested in this feature. I may just modify lndhub /getbalance to return lnd channelbalance since I am the only user of this hub (created my wallet then disabled /create)

**Comment by lndhub-admin** [comment]
maybe the new Admin Extension could have a component that makes a Admin wallet or where Admin can fund any account wallet.

the "LND won't let me pay an invoice created by itself," is to make a `payinvoice --allow_self_payment` call to LND that the Admin Extension component could do.

se : 

https://github.com/BlueWallet/LndHub/issues/196

https://github.com/BlueWallet/LndHub/pull/239

**Comment by lndhub-admin** [comment]
![Screenshot 2021-06-28 at 10 57 16](https://user-images.githubusercontent.com/86813532/124375062-52628b00-dca0-11eb-9ed7-898672c3662b.png)

**Comment by lndhub-admin** [comment]
https://github.com/lndhub-admin/LndHub-Admin-Extension

**Comment by xraid** [comment]
I am contemplating creating a __GRANT__ component using the __LndHub-Admin__ toolbox.

A component screen-card where accounts in system is showing in a table and where one can specify amount units and send.

The send is from __LndHub__ operator = admins __LND__ wallet funds.

When chosen a account(user) and specify amount click send after filling in the adminPin, the component will create a invoice from account(user) credentials and then that amount gets **Granted** to account(user) in the __LndHub__ Ledger.

Would the above suffice as functionality for case mentioned in this Request issue ?

**Comment by lndhub-admin** [comment]
aha ... also this triggered a associating problem of accounts being unnamed on the LndHub side
I now know howto get accounts named from a BlueWallet connected to a LndHub Admin enabled LND.
This without touching BlueWallet repo as a start and proof of concept, whereby later if find _tasty_ there could be a formalized add in BlueWallet also enabling and securing a sync of wallet account card "named" and its representation in LndHub ...

**Comment by junderw** [comment]
Any progress on this?

**Comment by jblachly** [comment]
Very much interested in this.

Use case / user story:

I have an LND node with, say, 1M sats. My child wants to be paid an allowance in sats, so I agree to transfer 10,000 sats monthly. Because we want to remain non-custodial, Bluewallet thru LNDHub makes sense. (Wallet of Satoshi, custodial; Phoenix and Muun non-custodial, but require new channels that will eat up huge fees).

LNDHub itself handles accounting just fine when the to/from are on the same node.

However, I have no way -- other than using on-chain activity or transfer out to a different Lightning Wallet then back in -- to fund an initial BlueWallet balance for myself.

What would be ideal is a command line admin tool to "instantiate" a balance, not greater than the sum of all un-allocated (in LNDHub's view) local channel funds, in an arbitrary LNDHub account.

I looked directly at the Redis keyspace to see if this were possible, but it appears not: Balance is computed directly from past lightning transactions.

**Question 1:** Is it possible to "synthesize" transactions to make this happen?
**Question 2:** The LNDHub schema documentation [1] describes a key, `balance_for_{userid} = {int}` but this does not appear in my Redis keyspace. Does this refer to on-chain balance? (I haven't made any on-chain transactions)


[1] https://github.com/BlueWallet/LndHub/blob/master/doc/schema.md

**Comment by xraid** [comment]
one can imagine a script that help in paying a invoice generated on account holders behalf or for edutainment let account holder create invoice in wallet, script would payout to invoice ...

same would go for case where one make invoice in wallet and pay oneself from LndHub and then pay invoice to other account

what script would do is either kust pay a invoice from LND or generate invoice and pay in one swoop ...
